src/main/java/dibo/Parser.java

```diff
@@ -0,0 +1,146 @@
+package dibo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import dibo.command.AddDeadlineCommand;
+import dibo.command.AddEventCommand;
+import dibo.command.AddToDoCommand;
+import dibo.command.ByeCommand;
+import dibo.command.Command;
+import dibo.command.DeleteCommand;
+import dibo.command.FindCommand;
+import dibo.command.ListCommand;
+import dibo.command.MarkCommand;
+import dibo.command.UnmarkCommand;
+import dibo.exception.DiboException;
+
+/**
+ * The class to make sense of the commands given by the user.
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    /**
+     * Takes in the command and tries to make sense of it.
+     * Returns a Command object which can be run at the end.
+     *
+     * @return A Command object to be run.
+     * @throws DiboException If an error occurs when parsing.
+     */
+    public static Command parse(String fullCommand) throws DiboException {
+        String[] commandDetails = fullCommand.split(" ");
+        switch (commandDetails[0]) {
```

Comment:
Maybe you can add the `break;` statement for all the cases or have a `// Fallthrough` comment?